frontend/src/notifications.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { Button } from "@chakra-ui/react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBellSlash } from "@fortawesome/free-solid-svg-icons";
+import { isEqual } from "lodash-es";
+
+type KeyType = number | string;
+
+type NotificationOptions<T> = {
+  filter: (x: T) => boolean;
+  message: (xs: T[]) => string;
+  key: (x: T) => KeyType;
+};
+
+export function useNotification<T>(
+  xs: T[],
+  { filter, message, key }: NotificationOptions<T>
+) {
+  const [seen, setSeen] = React.useState<KeyType[]>([]);
+  const filtered = xs.filter(filter);
+  const unseen = filtered.filter((x) => !seen.includes(key(x)));
```

Comment:
This seems N^2... which is probably fine because needing-QA and needing-CR are rarely bigger than 30 (30^2 == 900). Still a Set might be a better choice.

Reply:
:+1: 
If it becomes a problem, we can deal with it. That analysis is probably still over-pessimistic, because `seen` is `filtered` from the last run, so we're looking at `(filtered.length)^2`. And `filtered` is "pulls I should be notified about," so it should generally be much less than the maximum needing-CR or needing-QA number (like, if I've done ten CRs that are still on Pulldasher, then `filtered` should be ten items long). And 100 sounds totally reasonable to me.